app/com/arpnetworking/play/ProxyClient.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2015 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.play;
+
+import akka.util.ByteString;
+import com.google.inject.assistedinject.Assisted;
+import com.google.inject.assistedinject.AssistedInject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.libs.ws.WSClient;
+import play.shaded.ahc.org.asynchttpclient.AsyncHandler;
+import play.shaded.ahc.org.asynchttpclient.RequestBuilder;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A simple proxy client.
+ *
```

Comment:
I assume the copyright is Groupon because you copied this from another project. Can you link to that here?

Reply:
https://github.com/ArpNetworking/artemis/blob/master/app/client/ProxyClient.java

Artemis FTW!